clients/common/nm-meta-setting-desc.c

```diff
@@ -6661,7 +6661,6 @@ static const NMMetaPropertyInfo *const property_infos_OLPC_MESH[] = {
 	PROPERTY_INFO_WITH_DESC (NM_SETTING_OLPC_MESH_SSID,
 		.is_cli_option =                TRUE,
 		.property_alias =               "ssid",
-		.inf_flags =                    NM_META_PROPERTY_INF_FLAG_REQD,
```

Comment:
> For what seem to be historical reasons there's a default SSID.

where do you see a default? `verify()` in `libnm-core/nm-settings-olpc-mesh.c` seems to require a SSID.

Reply:
Hmmm. `nm-device-olpc-mesh.c`:

```
        if (!nm_setting_olpc_mesh_get_ssid (s_mesh)) {
                gs_unref_bytes GBytes *ssid = NULL;

                ssid = g_bytes_new_static (DEFAULT_SSID, NM_STRLEN (DEFAULT_SSID));
                g_object_set (G_OBJECT (s_mesh), NM_SETTING_OLPC_MESH_SSID, ssid, NULL);
        }
```

Perhaps it's stupid that there's a default SSID, but I guess it used to be the case in past and perhaps there still could be connections without a SSID?

No idea. What do you think? I'm fine also with dropping these lines.

Reply:
this code is in `complete_connection()`, which is only called during `AddAndActivate`, which in turn is called during `nmcli device connect ...`. That is fine but has no relevance for what constitutes a valid profile, because `complete_connection()` gets partial, invalid connections as input.

On the other hand, in `libnm-core/nm-setting-olpc-mesh.c` `verify()`, you see that a valid profile (currently) requires the SSID set, so nmcli also must require it. I think the patch should be dropped.

Reply:
Yes. Dropping it.